src/main/java/org/opentripplanner/apis/gtfs/GraphQLScalars.java

```diff
@@ -152,4 +336,138 @@ public Grams parseLiteral(Object input) throws CoercingParseLiteralException {
+          } else {
+            throw new CoercingSerializeException(
+              "Cannot serialize object of class %s as a ratio".formatted(
+                  dataFetcherResult.getClass().getSimpleName()
+                )
+            );
+          }
+        }
+
+        @Override
+        public Double parseValue(Object input) throws CoercingParseValueException {
+          if (input instanceof Double doubleValue) {
+            return validateRatio(doubleValue)
+              .orElseThrow(() ->
+                new CoercingParseValueException("Value is less than 0 or greater than 1.")
+              );
+          }
+          throw new CoercingParseValueException(
+            "Expected a number, got %s %s".formatted(input.getClass().getSimpleName(), input)
+          );
+        }
+
+        @Override
+        public Double parseLiteral(Object input) throws CoercingParseLiteralException {
+          var validationException = new CoercingParseLiteralException(
+            "Value is under 0 or greater than 1."
+          );
+          if (input instanceof FloatValue coordinate) {
+            return validateRatio(coordinate.getValue().doubleValue())
+              .orElseThrow(() -> validationException);
+          }
+          if (input instanceof IntValue coordinate) {
+            return validateRatio(coordinate.getValue().doubleValue())
+              .orElseThrow(() -> validationException);
+          }
+          throw new CoercingParseLiteralException(
+            "Expected a number, got: " + input.getClass().getSimpleName()
+          );
+        }
+
+        private static Optional<Double> validateRatio(double ratio) {
+          if (ratio >= -0.001 && ratio <= 1.001) {
+            return Optional.of(ratio);
+          }
+          return Optional.empty();
+        }
+      }
+    )
+    .build();
+
+  public static final GraphQLScalarType RELUCTANCE_SCALAR = GraphQLScalarType
+    .newScalar()
+    .name("Reluctance")
+    .coercing(
+      new Coercing<Double, Double>() {
+        private static final double MAX_Reluctance = 100000;
+
+        @Override
+        public Double serialize(@Nonnull Object dataFetcherResult)
+          throws CoercingSerializeException {
+          if (dataFetcherResult instanceof Double doubleValue) {
+            return doubleValue;
+          } else if (dataFetcherResult instanceof Float floatValue) {
+            return floatValue.doubleValue();
+          } else {
+            throw new CoercingSerializeException(
+              "Cannot serialize object of class %s as a reluctance".formatted(
+                  dataFetcherResult.getClass().getSimpleName()
+                )
+            );
+          }
+        }
+
+        @Override
+        public Double parseValue(Object input) throws CoercingParseValueException {
+          if (input instanceof Double doubleValue) {
+            if (Double.doubleToRawLongBits(doubleValue) < 0) {
```

Comment:
What should be the minimum value for reluctance? Does letting it be 0 ever make sense?

Reply:
We decided it should be 0.1 and I have changed the value.